src/ts-ast.ts

```diff
@@ -22,6 +25,25 @@ export class Document {
     this.members = data.members || [];
   }
 
+  /**
+   * Invoke `cb` on all nodes in this document, traversing depth-first.
+   */
+  walk(cb: (node: Node) => void) {
```

Comment:
What about instead defining `[Symbol.iterator]`?

Reply:
Similar, I converted this function into a generator. I also renamed it to `traverse` and hopefully clarified the description a bit.

The reason I'm not sure I want to make the node itself iterable is that I think it's not totally obvious whether iteration would be over the immediate children vs a full recursive walk. Hopefully the new name makes this clear.